keylime/agentstates.py

```diff
@@ -1,9 +1,47 @@
+import sys
 import threading
 
 from keylime.common.algorithms import Hash
 from keylime.ima.ast import get_FF_HASH, get_START_HASH
 from keylime.ima.file_signatures import ImaKeyrings
 
+if sys.version_info >= (3, 7):
+    from dataclasses import dataclass
+else:
+    from keylime.backport_dataclasses import dataclass
+
+
+@dataclass
+class TPMClockInfo:
+    clock: int
+    resetcount: int
+    restartcount: int
+    safe: int
+
+    @classmethod
+    def from_json(cls, data) -> "TPMClockInfo":
```

Comment:
Maybe rename to `from_dict` just for consistency reasons. I missed that last round.